### PR TITLE
Add Audit Service integration including schema, configurations, and A…

### PR DESCRIPTION
**Pull request by mushrafmim**
…PI gateway routes

## Summary
Provide a brief description of what this PR accomplishes and why these changes are needed.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made
- Added Audit Service

## Testing
- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [ ] I have tested edge cases
- [ ] All existing tests pass

## Checklist
- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues
Related to #5

### CODE REVIEW

File: init.sh [hunk 1]
```diff
@@ -466,6 +467,35 @@ fi
 print_success "Consent engine routes registered successfully"
 echo ""
 
+print_info "Exposing Required Audit Service Endpoints Publicly with OpenID Connect Authentication"
+
+curl --location --request PUT 'http://localhost:9180/apisix/admin/routes' \
+--header 'Content-Type: application/json' \
+--header 'X-API-KEY: QuNGwapKysRvHfUtNkQFbUaGiiYeOcGo' \
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply init.sh:474]: This is only for local testing, so, hard coding is ok

File: init.sh [hunk 2]
```diff
@@ -466,6 +467,35 @@ fi
...
+        "GET"
+    ],
+    "upstream": {
+        "type": "roundrobin",
+        "nodes": {
+            "audit-service:3001": 1
+        }
+    },
+    "id": "audit-endpoint"
+}
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply init.sh:488]: Removed Open ID Connect

File: ndx/.env [hunk 3]
```diff
@@ -114,4 +114,13 @@ IDP_ORG_NAME=Super
...
\ No newline at end of file
+IDP_JWKS_URL=http://wso2is:9764/oauth2/jwks
+
+# =============================================================================
+# AUDIT SERVICE CONFIGURATION
+# =============================================================================
+AUDIT_DB_HOST=postgres
+AUDIT_DB_PORT=5432
+AUDIT_DB_USERNAME=exchange
+AUDIT_DB_PASSWORD=1234
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply ndx/.env:126]: Local Development

File: ndx/audit_service_schema.sql [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    status character varying(20) NOT NULL,
+    event_type character varying(50),
+    event_action character varying(50),
+    actor_type character varying(50) NOT NULL,
+    actor_id character varying(255) NOT NULL,
+    target_type character varying(50) NOT NULL,
+    target_id character varying(255),
+    request_metadata text,
+    response_metadata text,
+    additional_metadata text,
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply ndx/audit_service_schema.sql:23]: Ok, I will consider

**Review by ginaxu1 — APPROVED** [review]

**Comment by ginaxu1** [comment]
is this really needed if audit service is optional? and eventually we replace `audit-service` in `opendif-core` with LSF standalone audit service?

**Review by sthanikan2000 — APPROVED** [review]

**Comment by mushrafmim** [comment]
> is this really needed if audit service is optional? and eventually we replace `audit-service` in `opendif-core` with LSF standalone audit service?

Yes, for now let's use this audit service inside the opendif-core and after we confirm that the `argus` is ready, let's move it to that.
